app/models/user.rb

```diff
@@ -1,2 +1,24 @@
 class User < ApplicationRecord
+  before_save :downcase_email
+
+  validates :name, presence: true, 
+    length: { maximum: 50 }
+    VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+
+  validates :email, presence: true, 
+    length: { maximum: 255 },
+    format: { with: VALID_EMAIL_REGEX },
+    uniqueness: true
+
+  validates :password, presence: true, 
+    length: { minimum: 6 }
+
+  has_secure_password
```

Comment:
I have 1 question for you, Please answer here :D
What does this line mean?

Reply:
Sorry,sir,which line? The empty line or?

Reply:
If for the has_secure_password,it will save the password and ecoded the password with command of user digest in rails console,must install gem bcrypt to activate it. During the installation, I also realise that the bcrypt .11 is corrupted and the invalid hash always appear,end up I reinstalled the later version to solve it